### PR TITLE
Add "Create React Native App" command to extension

### PR DESCRIPTION
**Pull request by dgomezc**
## PR checklist

- Quick summary of changes
   - Add "Create React Native App" command to extension
   - Update Extension-ReplaceNames.ps1 script
- Which issue does this PR relate to?
   - #1666

### CODE REVIEW

File: _build/Extension-ReplaceNames.ps1 [hunk 1]
```diff
@@ -19,10 +19,13 @@ Param(
   [string]$commandCategory,
 
   [Parameter(Mandatory = $True, Position = 7)]
-  [string]$wizardLaunchCommand,
+  [string]$createWebAppCommand,
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1:22]: Question: where and when should we add the deployReactNativeCommand? Probably when we implement that part?

File: _build/Extension-ReplaceNames.ps1 [hunk 2]
```diff
@@ -47,27 +50,30 @@ if ($vsixName) {
     $localDisplayName = $packagejsonContent.displayName
     $localVersionNumber = $packagejsonContent.version
     $localAiKey = $packagejsonContent.aiKey
-    $localwizardLaunchCommand = $packagejsonContent.contributes.commands[0].command
-    $localdeployAppCommand = $packagejsonContent.contributes.commands[1].command
+    $localcreateWebAppCommand = $packagejsonContent.contributes.commands[0].command
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]: picky picky one... sorry 
- local**D**eployWebAppCommand
- local**C**reateWebAppCommand
- local**C**reateReactNativeAppCommand

File: _build/Extension-ReplaceNames.ps1 [hunk 3]
```diff
@@ -47,27 +50,30 @@ if ($vsixName) {
...
+    $localdeployWebAppCommand = $packagejsonContent.contributes.commands[1].command
+    $localcreateReactNativeAppCommand = $packagejsonContent.contributes.commands[2].command
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    $localCreateWebAppCommand = $packagejsonContent.contributes.commands[0].command
    $localDeployWebAppCommand = $packagejsonContent.contributes.commands[1].command
    $localCreateReactNativeAppCommand = $packagejsonContent.contributes.commands[2].command
```

File: _build/Extension-ReplaceNames.ps1 [hunk 4]
```diff
@@ -47,27 +50,30 @@ if ($vsixName) {
...
 
     Write-Host "Replacing $LocalIdentity by  $vsixName"
     Write-Host "Replacing $localDisplayName by  $vsixDisplayName"
     Write-Host "Replacing $localVersionNumber by  $versionNumber"
-    Write-Host "Replacing $localwizardLaunchCommand by  $wizardLaunchCommand"
-    Write-Host "Replacing $localdeployAppCommand by  $deployAppCommand"
+    Write-Host "Replacing $localcreateWebAppCommand by  $createWebAppCommand"
+    Write-Host "Replacing $localdeployWebAppCommand by  $deployWebAppCommand"
+    Write-Host "Replacing $localcreateReactNativeAppCommand by  $createReactNativeAppCommand"
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    Write-Host "Replacing $localCreateWebAppCommand by  $createWebAppCommand"
    Write-Host "Replacing $localDeployWebAppCommand by  $deployWebAppCommand"
    Write-Host "Replacing $localCreateReactNativeAppCommand by  $createReactNativeAppCommand"
```

File: _build/Extension-ReplaceNames.ps1 [hunk 5]
```diff
@@ -47,27 +50,30 @@ if ($vsixName) {
...
 
     $content = $content -replace "$LocalIdentity" , "$vsixName"
     $content = $content -replace [regex]::Escape("$localDisplayName") , "$vsixDisplayName"
     $content = $content -replace "$localVersionNumber" , "$versionNumber"
     $content = $content -replace "$localAiKey" , "$aikey"
-    $content = $content -replace "$localwizardLaunchCommand" , "$wizardLaunchCommand"
-    $content = $content -replace "$localdeployAppCommand" , "$deployAppCommand"
+    $content = $content -replace "$localcreateWebAppCommand" , "$createWebAppCommand"
+    $content = $content -replace "$localdeployWebAppCommand" , "$deployWebAppCommand"
+    $content = $content -replace "$localcreateReactNativeAppCommand" , "$createReactNativeAppCommand"
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    $content = $content -replace "$localCreateWebAppCommand" , "$createWebAppCommand"
    $content = $content -replace "$localDeployWebAppCommand" , "$deployWebAppCommand"
    $content = $content -replace "$localCreateReactNativeAppCommand" , "$createReactNativeAppCommand"
```

File: _build/Extension-ReplaceNames.ps1 [hunk 6]
```diff
@@ -110,19 +116,22 @@ if ($vsixName) {
   # REPLACE commands IN extension.ts
   if (Test-Path($vsixExtensionTsFile)) {
 
-    $localWizardLaunchCommand = "webTemplateStudioExtension.wizardLaunch.local"
-    $localdeployAppCommand = "webTemplateStudioExtension.deployApp.local"
+    $localcreateWebAppCommand = "webTemplateStudioExtension.web.createApp.local"
+    $localdeployWebAppCommand = "webTemplateStudioExtension.web.deployApp.local"
+    $localcreateReactNativeAppCommand = "webTemplateStudioExtension.reactNative.createApp.local"
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    $localCreateWebAppCommand = "webTemplateStudioExtension.web.createApp.local"
    $localDeployWebAppCommand = "webTemplateStudioExtension.web.deployApp.local"
    $localCreateReactNativeAppCommand = "webTemplateStudioExtension.reactNative.createApp.local"
```

File: _build/Extension-ReplaceNames.ps1 [hunk 7]
```diff
@@ -110,19 +116,22 @@ if ($vsixName) {
...
+    Write-Host "Replacing $localcreateWebAppCommand by $createWebAppCommand in extension.ts"
+    Write-Host "Replacing $localdeployWebAppCommand by $deployWebAppCommand in extension.ts"
+    Write-Host "Replacing $localcreateReactNativeAppCommand by $createReactNativeAppCommand in extension.ts"
 
     $content = (Get-Content -path $vsixExtensionTsFile -Raw)
-    $content = $content -replace [regex]::Escape("$localWizardLaunchCommand") , "$wizardLaunchCommand"
-    $content = $content -replace [regex]::Escape("$localdeployAppCommand") , "$deployAppCommand"
+    $content = $content -replace [regex]::Escape("$localcreateWebAppCommand") , "$createWebAppCommand"
+    $content = $content -replace [regex]::Escape("$localdeployWebAppCommand") , "$deployWebAppCommand"
+    $content = $content -replace [regex]::Escape("$localcreateReactNativeAppCommand") , "$createReactNativeAppCommand"
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    $content = $content -replace [regex]::Escape("$localCreateWebAppCommand") , "$createWebAppCommand"
    $content = $content -replace [regex]::Escape("$localDeployWebAppCommand") , "$deployWebAppCommand"
    $content = $content -replace [regex]::Escape("$localCreateReactNativeAppCommand") , "$createReactNativeAppCommand"
```

File: _build/Extension-ReplaceNames.ps1 [hunk 8]
```diff
@@ -110,19 +116,22 @@ if ($vsixName) {
...
 
     [System.IO.File]::WriteAllLines($vsixExtensionTsFile, $content, [System.Text.UTF8Encoding]($False))
-    Write-Host "$resolvedPath - Commands applied ($localWizardLaunchCommand, $localdeployAppCommand)"
+    Write-Host "$resolvedPath - Commands applied ($localcreateWebAppCommand, $localdeployWebAppCommand, $localcreateReactNativeAppCommand)"
```
Comment (prupipho) [inline _build/Extension-ReplaceNames.ps1]:
```suggestion
    Write-Host "$resolvedPath - Commands applied ($localCreateWebAppCommand, $localDeployWebAppCommand, $localCreateReactNativeAppCommand)"
```

File: src/extension/src/extension.ts [hunk 9]
```diff
@@ -5,19 +5,15 @@ import { PLATFORM } from "./constants/constants";
...
-        await Deploy.getInstance(context).deployProject();
-      }
-    )
+    vscode.commands.registerCommand("webTemplateStudioExtension.web.createApp.local", async () => {
+      Controller.getInstance(context, PLATFORM.WEB);
+    }),
+    vscode.commands.registerCommand("webTemplateStudioExtension.reactNative.createApp.local", async () => {
+      Controller.getInstance(context, PLATFORM.REACTNATIVE);
+    }),
+    vscode.commands.registerCommand("webTemplateStudioExtension.deployApp.local", async () => {
```
Comment (prupipho) [inline src/extension/src/extension.ts:14]: is it missing **web.** here? on the command?
```suggestion
    vscode.commands.registerCommand("webTemplateStudioExtension.web.deployApp.local", async () => {
```

**Review by prupipho — DISMISSED** [review]
Left a few comments. Tested that still launch web local, react native local and installed version ok . 

Just doesn´t seem to support opening 2 local instances, one Web and one React Native at the same time

File: _build/Extension-ReplaceNames.ps1 (hunk 1 above)
Comment (prupipho) [reply _build/Extension-ReplaceNames.ps1:22]: though for web, not native apps. Not in the near future.

**Review by sibille — APPROVED** [review]

**Pull request merged**